File: pinyougou-shop-web/src/main/webapp/js/controller/sellerController.js
 //控制层 
app.controller('sellerController' ,function($scope,$controller,sellerService,loginService){	
	
	$controller('baseController',{$scope:$scope});//继承
	
    //读取列表数据绑定到表单中  
	$scope.findAll=function(){
		sellerService.findAll().success(
			function(response){
				$scope.list=response;
			}			
		);
	}    
	
	//分页
	$scope.findPage=function(page,rows){			
		sellerService.findPage(page,rows).success(
			function(response){
				$scope.list=response.rows;	
				$scope.paginationConf.totalItems=response.total;//更新总记录数
			}			
		);
	}
	
	//查询实体 
	$scope.findOne=function(id){				
		sellerService.findOne(id).success(
			function(response){
				$scope.entity= response;					
			}
		);				
	}
	
	//保存 
	$scope.save=function(){				
		var serviceObject;//服务层对象  				
		if($scope.entity.sellerId!=null){//如果有ID
			serviceObject=sellerService.update( $scope.entity ); //修改  
		}else{
			serviceObject=sellerService.add( $scope.entity  );//增加 
		}				
		serviceObject.success(
			function(response){
				if(response.success){
					//重新查询 
		        	$scope.reloadList();//重新加载
		        	alert(response.message);
				}else{
					alert(response.message);
				}
			}		
		);				
	}
	
	$scope.passwordList={sellerId:"",oldPassword:"",newPassword1:"",newPassword2:""};
	//修改密码 
	$scope.updatePassword=function(){
		if($scope.passwordList.newPassword1 == ""){
			return alert("新密码不能为空！");
		}
		if($scope.passwordList.newPassword1!=$scope.passwordList.newPassword2){
			return alert("新密码不一致！");
		}
		var serviceObject;//服务层对象  				
		if($scope.entity.sellerId!=null){//如果有ID
			$scope.passwordList.sellerId=$scope.entity.sellerId;
			serviceObject=sellerService.updatePassword($scope.passwordList); //修改  
		}else{
			alert("用户不存在！");
		}				
		serviceObject.success(
			function(response){				
				if(response.success){
					//重新查询 
					$scope.passwordList={};
		        	alert(response.message);		        	
				}else{
					alert(response.message);
				}
			}		
		);				
	}
	
	//新增
	$scope.add=function(){						
		sellerService.add($scope.entity).success(
			function(response){
				if(response.success){
					//如果注册成功跳转到登录页 
		        	location.href="shoplogin.html";
				}else{
					alert(response.message);
				}
			}		
		);				
	}
	
	 
	//批量删除 
	$scope.dele=function(){			
		//获取选中的复选框			
		sellerService.dele( $scope.selectIds ).success(
			function(response){
				if(response.success){
					$scope.reloadList();//刷新列表
					$scope.selectIds=[];
				}						
			}		
		);				
	}
	
	$scope.searchEntity={};//定义搜索对象 
	
	//搜索
	$scope.search=function(page,rows){			
		sellerService.search(page,rows,$scope.searchEntity).success(
			function(response){
				$scope.list=response.rows;	
				$scope.paginationConf.totalItems=response.total;//更新总记录数
			}			
		);
	}
	

	//获取当前用户名和商家信息
	$scope.findLoginUser = function(){
		loginService.loginName().success(
				function(response){
					$scope.loginName = response.loginName;
					//获取商家信息
					$scope.findOne($scope.loginName);
				}
		);		
	}
	
    
});	

File: pinyougou-shop-web/src/main/java/com/pinyougou/shop/controller/SellerController.java
package com.pinyougou.shop.controller;
import java.util.HashMap;
import java.util.List;

import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;
import com.alibaba.dubbo.config.annotation.Reference;
import com.pinyougou.pojo.TbSeller;
import com.pinyougou.sellergoods.service.SellerService;

import entity.PageResult;
import entity.Result;
/**
 * controller
 * @author Administrator
 *
 */
@RestController
@RequestMapping("/seller")
public class SellerController {

	@Reference
	private SellerService sellerService;
	
	/**
	 * 返回全部列表
	 * @return
	 */
	@RequestMapping("/findAll")
	public List<TbSeller> findAll(){			
		return sellerService.findAll();
	}
	
	
	/**
	 * 返回全部列表
	 * @return
	 */
	@RequestMapping("/findPage")
	public PageResult  findPage(int page,int rows){			
		return sellerService.findPage(page, rows);
	}
	
	/**
	 * 增加
	 * @param seller
	 * @return
	 */
	@RequestMapping("/add")
	public Result add(@RequestBody TbSeller seller){
		//密码加密
	    BCryptPasswordEncoder passwordEncoder = new BCryptPasswordEncoder();
		String password = passwordEncoder.encode(seller.getPassword());//加密
		seller.setPassword(password);
		try {
			sellerService.add(seller);
			return new Result(true, "增加成功");
		} catch (Exception e) {
			e.printStackTrace();
			return new Result(false, "增加失败");
		}
	}
	
	/**
	 * 修改
	 * @param seller
	 * @return
	 */
	@RequestMapping("/update")
	public Result update(@RequestBody TbSeller seller){
		try {
			sellerService.update(seller);
			return new Result(true, "修改成功");
		} catch (Exception e) {
			e.printStackTrace();
			return new Result(false, "修改失败");
		}
	}
	
	/**
	 * 更新用户密码
	 * @param seller
	 * @return
	 */
	@RequestMapping("/updatePassword")
	public Result updatePassword(@RequestBody HashMap<String,String> passwordList){	    
	  
	    TbSeller seller = sellerService.findOne(passwordList.get("sellerId"));
	  //密码加密
	    BCryptPasswordEncoder passwordEncoder = new BCryptPasswordEncoder();
	    String oldPassword = passwordList.get("oldPassword");
	    String newPassword1 = passwordList.get("newPassword1");
	    String newPassword2 = passwordList.get("newPassword2");
	    String newPassword="";
	   if(newPassword1.equals(newPassword2)){
	       newPassword= passwordEncoder.encode(newPassword1);
	   }else{
	       return new Result(false, "新密码不一致");  
	   }
	   if(passwordEncoder.matches(oldPassword, seller.getPassword())){
	       seller.setPassword(newPassword);
	   }else{
	       return new Result(false, "原密码错误，修改失败。");
	   }

       try {
           sellerService.update(seller);
           return new Result(true, "修改成功");
       } catch (Exception e) {
           e.printStackTrace();
           return new Result(false, "修改失败");
       }
	}
	
	/**
	 * 获取实体
	 * @param id
	 * @return
	 */
	@RequestMapping("/findOne")
	public TbSeller findOne(String id){
		return sellerService.findOne(id);		
	}
	
	/**
	 * 批量删除
	 * @param ids
	 * @return
	 */
	@RequestMapping("/delete")
	public Result delete(String [] ids){
		try {
			sellerService.delete(ids);
			return new Result(true, "删除成功"); 
		} catch (Exception e) {
			e.printStackTrace();
			return new Result(false, "删除失败");
		}
	}
	
		/**
	 * 查询+分页
	 * @param brand
	 * @param page
	 * @param rows
	 * @return
	 */
	@RequestMapping("/search")
	public PageResult search(@RequestBody TbSeller seller, int page, int rows  ){
		return sellerService.findPage(seller, page, rows);		
	}
	
	
	
}
